modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/TransportSearchTemplateAction.java

```diff
@@ -110,6 +111,19 @@ static SearchRequest convert(SearchTemplateRequest searchTemplateRequest, Search
             builder.parseXContent(parser, false);
             builder.explain(searchTemplateRequest.isExplain());
             builder.profile(searchTemplateRequest.isProfile());
+            if (searchRequest.source() == null) {
+                searchRequest.source(new SearchSourceBuilder());
+            }
+            Integer trackTotalHitsUpTo = searchRequest.source().trackTotalHitsUpTo();
+            if (trackTotalHitsUpTo != null) {
```

Comment:
Can you add a comment that a non-value of trackTotalHitsUpTo here indicates that `rest_total_hits_as_int` was used ?